### PR TITLE
Upgraded concatenate filter to ES 5.4.2 and added unit tests for main…

### PR DESCRIPTION
**Pull request by maoning**
- Upgraded concatenate filter to ES 5.4.2 
- Added unit tests using existing elasticsearch testing framework

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,14 +2,17 @@
 Elasticsearch plugin which only provides a TokenFilter that merges tokens in a token stream back into one. Taken from http://elasticsearch-users.115913.n3.nabble.com/Is-there-a-concatenation-filter-td3711094.html
 
 ## ElasticSearch version support
-This plugin targets 1.X versions of ES, and won't work for 2.X.
+This plugin is compatible with ES 5.4.2. The original forked version is at https://github.com/francesconero/elasticsearch-concatenate-token-filter
```
Comment (adrianocrestani) [inline README.md]: We plan to create a merge request against that fork, therefore, that extra info will not be necessary anymore. No?

File: pom.xml [hunk 2]
```diff
@@ -5,67 +5,69 @@
 	<modelVersion>4.0.0</modelVersion>
 	<groupId>elasticsearch.concatenate</groupId>
 	<artifactId>elasticsearch-concatenate</artifactId>
-	<version>1.1.0</version>
+	<version>5.4.2</version>
```
Comment (jpdaigle) [inline pom.xml]: Is that the plugin's version? Is it typical for community / 3rdparty plugins to list their version as exactly the ES version?

File: src/main/resources/plugin-descriptor.properties [hunk 3]
```diff
@@ -0,0 +1,6 @@
+description=${project.description}
+version=${project.version}
+name=${project.artifactId}
+java.version=1.8
```
Comment (jpdaigle) [inline src/main/resources/plugin-descriptor.properties:4]: is that the same as what ES requires? We don't want to cause a deployment problem by having stricter requirements than ES itself

File: README.md (hunk 1 above)
Comment (maoning) [reply README.md]: Make sense, I have removed the unnecessary reference.

File: pom.xml (hunk 2 above)
Comment (maoning) [reply pom.xml]: You are right, it is the plugin version, which is not related to ES version. I have updated it 1.2.0 instead.

File: src/main/resources/plugin-descriptor.properties (hunk 3 above)
Comment (maoning) [reply src/main/resources/plugin-descriptor.properties:4]: I have checked ES official document, these fields are required for newer ES plugins: https://www.elastic.co/guide/en/elasticsearch/plugins/5.4/plugin-authors.html

Comment (jpdaigle) [reply src/main/resources/plugin-descriptor.properties:4]: Yes, sorry I didn't say what I meant, which was "is the minimum version of Java 1.8 a new requirement compared to base ES 5.4?". It turns out to be fine (ES 5.4 already requires 1.8), so I withdraw the original comment 😄

**Review by jpdaigle — APPROVED** [review]

**Review by adrianocrestani — APPROVED** [review]

**Pull request merged**